web-gateway/src/main/java/controllers/TransactionController.java

```diff
@@ -77,28 +78,59 @@ public static Call transactionsPage(TransactionInfoStatus status, int page, int
                 loadNav(user).thenComposeAsync(nav -> {
-                    UUID itemId = UUID.fromString(id);
-                    CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
-                    return transactionFuture.handle((transaction, exception) -> {
-                        if (exception == null) {
-                            Optional<User> seller = Optional.empty();
-                            Optional<User> winner = Optional.empty();
-                            for (User u : nav.getUsers()) {
-                                if (transaction.getCreator().equals(u.getId())) {
-                                    seller = Optional.of(u);
-                                }
-                                if (transaction.getWinner().equals(u.getId())) {
-                                    winner = Optional.of(u);
+                            UUID itemId = UUID.fromString(id);
+                            CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
+
+                           return transactionFuture.handle((transaction, exception) -> {
+                               Optional<User> seller = null;
+                               Optional<User> winner = null;
+                               if (exception == null) {
+
+
+                                   seller = Optional.empty();
+                                   winner = Optional.empty();
+
+                                   try {
+                                       if (seller.isPresent() ) {
+
+
+                                           seller = Optional.of(userService.getUser(transaction.getCreator()).invoke().toCompletableFuture().get());
+                                       }
+                                   else{
+                                           CompletableFuture.completedFuture(redirect(routes.UserController.createUserForm()));
+                                       }
+                                   } catch (InterruptedException e) {
+
+                                   } catch (ExecutionException e) {
+
+                                   }
+
+
+                                   try {
+                                       if(winner.isPresent()) {
+                                           winner = Optional.of(userService.getUser(transaction.getWinner()).invoke().toCompletableFuture().get());
+                                       }
+                                       else{
+                                           CompletableFuture.completedFuture(redirect(routes.UserController.createUserForm()));
+                                       }
+                                   } catch (InterruptedException e) {
+
+                                   } catch (ExecutionException e) {
+
+                                   }
```

Comment:
This code has the same problems than `ItemController`. I'm beginning to think that this needs a deeper review: maybe extract common code to `AbstractController`, maybe creating a `UserLoader` which may load many users in parallel ( `UserLodaer.load(UUID... uuid) -> Map<UUID,User>`), or some other way to reuse this code. There are 30 lines of code polluting our main logic and they _just_  load 2 fields.